File: view/src/test/java/by/rekuts/travelagency/view/ViewTestConfig.java
package by.rekuts.travelagency.view;

import by.rekuts.travelagency.service.*;
import by.rekuts.travelagency.view.controller.CreateController;
import by.rekuts.travelagency.view.controller.GetController;
import org.mockito.Mockito;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.config.annotation.EnableWebMvc;

@Configuration
@EnableWebMvc
public class ViewTestConfig {
    @Bean
    public TourService tourService() {
        return Mockito.mock(TourService.class);
    }

    @Bean
    public CountryService countryService() {
        return Mockito.mock(CountryService.class);
    }

    @Bean
    public UserService userService() {
        return Mockito.mock(UserService.class);
    }

    @Bean
    public ReviewService reviewService() {
        return Mockito.mock(ReviewService.class);
    }

    @Bean
    public HotelService hotelService() {
        return Mockito.mock(HotelService.class);
    }

    @Bean
    public GetController mainController() {
        return new GetController();
    }

    @Bean
    public CreateController createController() {
        return new CreateController();
    }
}

File: view/src/test/scala/by/rekuts/travelagency/view/configuration/ViewTestConfig.scala
package by.rekuts.travelagency.view.configuration

import by.rekuts.travelagency.service._
import by.rekuts.travelagency.view.controller._
import org.mockito.Mockito
import org.springframework.context.annotation.{Bean, Configuration}
import org.springframework.web.multipart.support.StandardServletMultipartResolver
import org.springframework.web.servlet.config.annotation.EnableWebMvc

@Configuration
@EnableWebMvc
class ViewTestConfig {
  @Bean
  def tourService: TourService = Mockito.mock(classOf[TourService])

  @Bean
  def countryService: CountryService = Mockito.mock(classOf[CountryService])

  @Bean
  def userService: UserService = Mockito.mock(classOf[UserService])

  @Bean
  def reviewService: ReviewService = Mockito.mock(classOf[ReviewService])

  @Bean
  def hotelService: HotelService = Mockito.mock(classOf[HotelService])

  @Bean
  def getController = new GetController(tourService, countryService, userService, reviewService, hotelService)

  @Bean
  def createController = new CreateController(userService, countryService, tourService, reviewService, hotelService)

  @Bean
  def deleteController = new DeleteController(reviewService, countryService, tourService, hotelService, userService)

  @Bean
  def updateController = new UpdateController(countryService, hotelService, reviewService, tourService, userService)

  @Bean
  def importController = new ImportController(tourService)

  @Bean
  def multipartResolver = new StandardServletMultipartResolver
}
